ergo-lib/src/wallet/box_selector/simple.rs

```diff
@@ -45,7 +45,8 @@ impl<T: ErgoBoxAssets> BoxSelector<T> for SimpleBoxSelector {
         let mut selected_boxes_value: u64 = 0;
         let target_balance: u64 = target_balance.into();
         // sum all target tokens into hash map (think repeating token ids)
-        let mut target_tokens_left: HashMap<TokenId, TokenAmount> = sum_tokens(Some(target_tokens));
+        let mut target_tokens_left: HashMap<TokenId, TokenAmount> =
+            sum_tokens(Some(target_tokens)).map_err(BoxSelectorError::TokenAmountError)?;
```

Comment:
It can be simplified to:
```suggestion
            sum_tokens(Some(target_tokens))?;
```
if `From` is implemented like in https://github.com/ergoplatform/sigma-rust/blob/86e012029ffab9590ef0898ff98f11bd30b8b004/ergotree-interpreter/src/eval.rs#L137 (with `use thiserror::Error;
` imported).

Reply:
Nice thanks!